### ISSUE TITLE
Unable to deploy databricks withouth subscription permissions

### ISSUE THREAD

**Issue by RHooijer** [hashicorp/terraform-provider-azurerm#18382]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.9

### AzureRM Provider Version

3.2.2

### Affected Resource(s)/Data Source(s)

azurerm_databricks_workspace

### Terraform Configuration Files

```hcl
resource "azurerm_databricks_workspace" "this" {
  name                = var.name
  resource_group_name = var.resource_group_name
  location            = var.location
  
  infrastructure_encryption_enabled = true
  #public_network_access_enabled = false
  network_security_group_rules_required = "AllRules"
  sku                 = var.sku  

  custom_parameters {
    virtual_network_id                                   = var.virtual_network_id
    public_subnet_name                                   = var.subnet_name_host
    private_subnet_name                                  = var.subnet_name_container
    public_subnet_network_security_group_association_id  = var.subnet_host_network_security_group_association_id
    private_subnet_network_security_group_association_id = var.subnet_container_network_security_group_association_id 
    #no_public_ip                                         = true
  }

  lifecycle {
    ignore_changes = [
      tags
    ]
  }
}
```

### Debug Output/Panic Output

```shell
performing CreateOrUpdate: workspaces.WorkspacesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="AuthorizationFailed" Message="The client '' with object id '' does not have authorization to perform action 'Microsoft.ManagedIdentity/register/action' over scope '/subscriptions/' or the scope is invalid. If access was recently granted, please refresh your credentials."
```

### Expected Behaviour

Databricks is deployed

### Actual Behaviour

Databricks is not deployed

### Steps to Reproduce

Deploy databricks in an existing resoruce group without permissions on the subscription level. 

### Important Factoids

na

### References

na

**Comment by aristosvo** [comment]
Hi @RHooijer ! 

Thanks for filing this issue, could you elaborate a bit on the setup? Does your service principal/user have `User Access Administrator` rights on both VNet resource group and its own resource group (can be the same..)?

If that is already the case I don't have a clue yet, could you in that case expand the setup with your VNet configuration to ease reproduction? Thanks in advance!

**Comment by RHooijer** [comment]
Hi @aristosvo, There has been a change at Microsoft & Databricks side. When deploying an Databricks workspace, the managed identity provider is register at the subscription level. However, there was an issue that even when this was already done, the Azure RM deployment still wanted to register the provider, since our Service Principal doesn't had subscription level permissions it failed. I though it was fixed by going to an older terraform version, but it seems that while I switched Microsoft fixed the issue (confirmed by MS Support). The issue therefore has already been solved. Thanks for taking the time to respond though, much appreciated.
